sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonUtils.scala

```diff
@@ -55,4 +60,26 @@ object JacksonUtils {
 
     schema.foreach(field => verifyType(field.name, field.dataType))
   }
+
+  private def validateStringLiteral(exp: Expression): String = exp match {
+    case Literal(s, StringType) => s.toString
+    case e => throw new AnalysisException("Must be a string literal, but: " + e)
+  }
+
+  def validateSchemaLiteral(exp: Expression): StructType =
+    DataType.fromJson(validateStringLiteral(exp)).asInstanceOf[StructType]
```

Comment:
Ah, thanks. Yes, if it throws a class cast exception, I think we should produce a better exception and message rather than just one saying `A cannot be cast to B`. Maybe, add a util for both places?